R/ari_spin.R

```diff
@@ -57,115 +49,107 @@
   images <- normalizePath(images)
   output_dir <- normalizePath(dirname(output))
-
+  # Paragraphs
   if (length(paragraphs) == 1) {
     if (file.exists(paragraphs)) {
       paragraphs <- readLines(paragraphs, warn = FALSE)
       paragraphs <- paragraphs[!paragraphs %in% ""]
     }
   }
-
+  # Paragraphs: Check for semicolons
   semi_colon <- trimws(paragraphs) == ";"
   if (any(semi_colon)) {
     warning(paste0(
       "Some paragraphs are simply a semicolon - ",
       "likely needs to be replaced or slide removed!"
     ))
   }
+  # Check for arguments
   stopifnot(
     length(paragraphs) > 0,
     identical(length(images), length(paragraphs)),
     all(file.exists(images)),
     dir.exists(output_dir)
   )
-  # End of Argument checks
 
   # Setup objects to populate in for-loop with tts()
   wave_objects <- vector(mode = "list", length = length(paragraphs))
-  par_along <- seq_along(paragraphs)
+  paragraphs_along <- seq_along(paragraphs)
   ideal_duration <- rep(NA, length(paragraphs))
 
-  pb <- progress_bar$new(
-    format = "Fetching Narration [:bar] :percent",
-    total = length(par_along)
-  )
   # Iterate through arguments used in tts()
-  for (i in par_along) {
-    args <- tts_args
-    args$text <- paragraphs[i]
-    args$voice <- voice
-    args$service <- service
+  for (ii in paragraphs_along) {
+    args <- tts_engine_args
+    args$text <- paragraphs[ii]
     args$bind_audio <- TRUE
     # coqui+ari doesn't work with mp3
-    if (service == "coqui") {
+    if (tts_engine_args$service == "coqui") {
       args$output_format <- "wav"
+      args$voice <- NULL
     }
-    wav <- do.call(text2speech::tts, args = args)
+    wav <- do.call(tts_engine, args = args)
     wav <- reduce(wav$wav, bind)
-    wav <- pad_wav(wav, duration = duration[i])
-    ideal_duration[i] <- length(wav@left) / wav@samp.rate
-    wave_objects[[i]] <- wav
-    pb$tick()
-  }
-  if (service == "coqui") {
-    cli::cli_alert_info("Coqui TTS does not support MP3 format; will produce a WAV audio output.")
+    wav <- pad_wav(wav, duration = duration[ii])
+    ideal_duration[ii] <- length(wav@left) / wav@samp.rate
+    wave_objects[[ii]] <- wav
   }
-
   # Burn subtitles
   if (subtitles) {
     sub_file <- paste0(file_path_sans_ext(output), ".srt")
     ari_subtitles(paragraphs, wave_objects, sub_file)
   }
-
+  print("Audio succesfully converted...............")
   # Create a video from images and audio
-  res <- ari_stitch(images, wave_objects, output, ...)
-
+  res <- ari_stitch(images, wave_objects, output)
   # Collect output
-  args <- list(...)
+  args <- list()
   cleanup <- args$cleanup
   if (is.null(cleanup)) {
     cleanup <- TRUE
   }
   if (!cleanup) {
     attr(res, "wavs") <- wave_objects
   }
-  attr(res, "voice") <- voice
+  attr(res, "voice") <- tts_engine_args$voice
   if (subtitles) {
     attr(res, "subtitles") <- sub_file
   }
-  attr(res, "service") <- service
+  attr(res, "service") <- tts_engine_args$service
   return(res)
 }
```

Comment:
We must remove this attributes situation, this function should return a path to a video file.